api/v1/checks.go

```diff
@@ -563,8 +563,13 @@ type JunitCheck struct {
 	// namespace in which canary is created
 	namespace string `yaml:"-" json:"-"`
 	// name of the canary. will be the same for the pod
-	name string     `yaml:"-" json:"-"`
-	Spec v1.PodSpec `yaml:"spec" json:"spec"`
+	name string `yaml:"-" json:"-"`
+	// interval of the canary
+	interval uint64 `yaml:"-" json:"-"`
+	//timeout in minutes to wait for specified container to finish its job.
```

Comment:
Maybe we just inject the canary back in ?

` parent Canary`

Reply:
What do you mean?